File: src/backend/errors/error_content.py
import json
from errors.exceptions import BaseAppException
from constants.env_variables import AppConfig

class ErrorContent:
    """
    A Python version of your TypeScript "ErrorContent" concept.
    It takes an exception and builds a structured dict for JSON responses.
    """

    def __init__(self, *, status: int, code: str, message: str, details: dict):
        self.type = "Error"
        self.target = AppConfig.APP_NAME
        self.status = status
        self.code = code
        self.message = message
        self.ext = details 
        self.raw = {}

    @classmethod
    def from_exception(cls, exc: BaseAppException) -> "ErrorContent":
        """
        Build an ErrorContent from one of our custom exceptions.
        """
        return cls(
            status=exc.status,
            code=exc.code if exc.code else "UNKNOWN",
            message=exc.message,
            details=exc.details
        )

    def get_summary(self) -> dict:
        """
        Return only the main fields without raw details.
        """
        return {
            "type": self.type,
            "target": self.target,
            "status": self.status,
            "code": self.code,
            "message": self.message,
            "ext": self.ext
        }

    def get_details(self) -> dict:
        """
        Include raw as well.
        """
        data = self.get_summary()
        data["raw"] = self.raw
        return data

    def to_json_str(self) -> str:
        """
        Return the full details as a JSON string, if needed.
        """
        return json.dumps(self.get_details())

File: src/server/types/errorContent.js
import {
  HttpRequestException,
  BadRequestException,
  UnauthorizedException,
  ForbiddenException,
  Exception
} from './exceptions.js';
import _ from 'lodash';

export class ErrorContent {
  constructor(desc) {
    this.type = '';
    this.target = 'SyntaxSentinals';
    this.status = desc[0];
    this.code = '';
    this.message = desc[1];
    this.ext = {};
    this.raw = desc[2];
  }

  static convertFromException(exception) {
    const error = new ErrorContent([500, exception ? exception.message : 'Unknown error', {}]);

    if (exception instanceof BadRequestException) {
      error.status = 400;
    }
    if (exception instanceof UnauthorizedException) {
      error.status = 401;
    }
    if (exception instanceof ForbiddenException) {
      error.status = 403;
    }
    if (exception instanceof HttpRequestException) {
      error.status = exception.status ?? 500;
      error.ext = exception.details ?? {};
      error.code = exception.code;
    }
    if (exception instanceof Error) {
      error.status = 500;
    }
    return error;
  }

  getSummary() {
    const properties = ['type', 'target', 'status', 'code', 'message', 'ext'];
    return _.pick(this, properties);
  }

  getDetails() {
    const properties = ['type', 'target', 'status', 'code', 'message', 'ext', 'raw'];
    return _.pick(this, properties);
  }

  static convertFromExcToError(exception) {
    const errorContent = ErrorContent.convertFromException(exception);
    const error = new Error(JSON.stringify(errorContent.getDetails()));
    return error;
  }
}